### PR TITLE
Different ques based on bin_search

### PR DESCRIPTION
**Pull request by hinduBale**
Interview based question to improve time complexity from O(n^2) to O(nlog n)

### CODE REVIEW

File: Binary_search/test.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+void dfsForAll(vector <int> graph[],int v)
+{
+    vector<bool> visited(v, false);
+    for(int u = 0; u < v; u++)
+        if(visited[u] == false)
+            dfs(graph, u, visited);
+}
+
+int main()
```
Comment (starkblaze01) [inline Binary_search/test.cpp]: Hi, @hinduBale thanks for PR, can you tell me what is this main function executing?

Comment (hinduBale) [reply Binary_search/test.cpp]: Ohh, I mistakenly pushed this dfs file in the binary search folder. I'll quickly remove that... It's not even a actual running program, I was just testing something!!

Comment (starkblaze01) [reply Binary_search/test.cpp]: Any progress?

File: Binary_search/search_in_rotated_array.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                return findRotEle(arr, mid+1, r, key);
+        }
+    }
+    return -1;
+}
+
+int main()
+{
+    lli test;
+    cin >> test;
```
Comment (starkblaze01) [inline Binary_search/search_in_rotated_array.cpp]: Hi, @hinduBale thanks for making changes. Can you add a comment in this file also which inputs are being taken, or print it on the console?

File: Binary_search/flyBinSearch.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+//    }
+//    return -1;
+//}
+//
+//int main()
+//{
+//    int A[13] = {6,7,8,9,10,11,12,13,14,15,16,17,18};
+//    cout << binSearch(A, 0, 12, 17) << endl;
+//    return 0;
+//}
```
Comment (starkblaze01) [inline Binary_search/flyBinSearch.cpp]: Please remove this comment also.

**Pull request merged**